pylint: Update pylint to 2.4.4

chu11 opened the pull request:
Per #3034, update pylint from 2.2.2 to 2.4.4 to get around `isort` module errors.  Update python to handle new pylint errors.

Most of the Python changes should be simple and obvious.  Perhaps the only one to consider was the `invalid-name` fixes.  New variable names acceptable?  I chose to disable `invalid-name` for one function parameter (`raiseJobException`), which seemed quite specific.  I did change the variable name for another function parameter (`io` -> `iotype`).

Also, i chose to ignore `too-many-lines` errorrs.  Could increase the max-lines instead (default 1000)?

grondo commented:
>  New variable names acceptable? 

Can we update the `pylint/.pylintrc` instead of changing names? The pylint "acceptable variable names" rules seem inane and to me and have nothing to do with correctness or code readability in the spots we've "violated" them IMO.

I agree also the "too many lines" errors are a bit ridiculous. Can we also turn that off in `.pylintrc`? (at least for now?)

chu11 commented:
> I agree also the "too many lines" errors are a bit ridiculous. Can we also turn that off in .pylintrc? (at least for now?)

Yup, already did that (vs increasing the max-length config).

chu11 commented:
re-pushed, already squashing in changes since they are so minor:

- adjusted pylintrc to support `io` and `rc` variables
- adjusted pylintrc to support upper case characters in variable names, attribute names, and function arguments
- reverted my `rc` -> `retval` change.  BUT I left my `io` -> `iotype` change from before
- reverted my disabling of `invalid-name` for the variable `raiseJobException`, which is now legal.

grondo commented:
Ok feel free to set MWP.

chu11 commented on src/bindings/python/flux/job.py:
```diff
@@ -386,9 +385,8 @@ def get_event(self, autoreset=True):
-                # re-raise all other exceptions
-                raise
+            # re-raise all other exceptions
+            raise
```
My intuition was it was the same as:

```
if SOMETHING:
    return 0
else:
    return 1
```

I'm very used to removing the `else` in something like the above.  Perhaps I'm just not used to seeing the same issue with `raise`?

grondo replied:
Maybe that is it, it doesn't bother me that much. I guess in this case pylint is just enforcing an opinion. Do we really *require* pylint's opinion being enforced here. I guess the one benefit is consistency of code. 

Sorry, if you can't tell I'm only slightly annoyed that pylint unnecessarily wasted a portion of both our mornings  :laughing:

garlick commented:
Thanks so much for jumping on this @chu11!

grondo commented:
Ok to set MWP?

grondo commented:
BTW, dots (`.`) are allowed in git branch names, fyi. :laughing: :stuck_out_tongue_winking_eye:

SteVwonder commented on src/bindings/python/flux/core/handle.py:
```diff
@@ -155,18 +158,12 @@ def msg_watcher_create(
         return TimerWatcher(self, after, callback, repeat=repeat, args=args)
 
     def signal_watcher_create(self, signum, callback, args=None):
-        from flux.core.watchers import SignalWatcher
```
TBH, I am not totally sure.  I haven't been able to find a discussion about any other factors besides the trade-off between "code cleanliness" and  efficiency.  I suspect what happened is that for 99.99% of all use cases, the performance between the two import styles is in the noise, and so the simplicity/cleanliness side is what got enshrined in [PEP8](https://www.python.org/dev/peps/pep-0008/#imports). There may also be an argument for "failing fast" when you don't have a required dependency.

grondo replied:
Ah, failing fast is actually a pretty good reason.
